mlflow/tracking/_tracking_service/client.py

```diff
@@ -753,7 +753,12 @@ def _record_logged_model(self, run_id, mlflow_model):
         self.store.record_logged_model(run_id, mlflow_model)
 
     def _get_artifact_repo_for_trace(self, trace_info: TraceInfo):
-        artifact_uri = next(v for k, v in trace_info.tags.items() if k == "mlflow.artifactLocation")
+        if MLFLOW_ARTIFACT_LOCATION not in trace_info.tags:
+            raise MlflowException(
+                "Trace artifact location not specified, please specify it with "
+                f"tag '{MLFLOW_ARTIFACT_LOCATION}' in the trace."
+            )
+        artifact_uri = trace_info.tags[MLFLOW_ARTIFACT_LOCATION]
```

Comment:
We might want to make this tag immutable to prevent security vulnerabilities.

Reply:
Oh yea now we should exclude it for all stores